### PR TITLE
golem cli

### LINKED ISSUE #439: Golem CLI for provider.
https://www.notion.so/golemnetwork/Golem-CLI-Tree-Draft-v-0-1-9c373365ffd64263ad31b56e2c1be0dc

### PR DESCRIPTION
**Pull request by etam**
Resolves #439
Requires https://github.com/golemfactory/ya-client/pull/45

See individual commits for some clarity.

### CODE REVIEW

**Comment by kubkon** [comment]
FYI #513 cause this might cause some merge conflicts down the line ;-)

**Comment by etam** [comment]
> FYI #513 cause this might cause some merge conflicts down the line ;-)

Peace was never an option. We'll have to resolve it by duel ;)
Thanks for info.

File: golem_cli/src/service.rs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+async fn reader_to_log<T: tokio::io::AsyncRead + Unpin>(name: String, reader: T) {
+    let mut reader = tokio::io::BufReader::new(reader);
+    let mut buf = Vec::new();
+    loop {
+        match reader.read_until(b'\n', &mut buf).await {
+            Ok(len) => {
+                if len > 0 {
+                    log::debug!(
```
Comment (mfranciszkiewicz) [inline golem_cli/src/service.rs:40]: I'd suggest redirecting daemon logs to a file and displaying provider logs with their respective log level (preserving colored outputs)

**Review by mfranciszkiewicz — APPROVED** [review]
I guess the next step is to move `golem_cli` code into `ya-provider`

File: golem_cli/src/service.rs (hunk 1 above)
Comment (etam) [reply golem_cli/src/service.rs:40]: There's a `TODO` for this in line 67

Comment (mfranciszkiewicz) [reply golem_cli/src/service.rs:40]: Yes, and it's rather vague :slightly_smiling_face:

**Review by mfranciszkiewicz — APPROVED** [review]

File: golem_cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "golem"
```
Comment (tworec) [inline golem_cli/Cargo.toml:2]:
```suggestion
name = "ya-cli"
```

Comment (tworec) [reply golem_cli/Cargo.toml:2]: I'd suggest renaming also the directory for this crate

**Pull request merged**